### PR TITLE
[VL] Support casting varchar type to timestamp type

### PR DESCRIPTION
**Pull request by PHILO-HE**
## What changes were proposed in this pull request?

Velox pr: https://github.com/facebookincubator/velox/pull/11975

## How was this patch tested?

Existing test.

### CODE REVIEW

**Comment by PHILO-HE** [comment]
There are only few corner cases that got inconsistent result. The excluded test requiring further fix is tracked by https://github.com/apache/incubator-gluten/issues/8984.

File: cpp/velox/substrait/SubstraitToVeloxPlanValidator.cc [hunk 1]
```diff
@@ -262,7 +262,7 @@ bool SubstraitToVeloxPlanValidator::isAllowedCast(const TypePtr& fromType, const
   }
 
   // Limited support for X to Timestamp.
-  if (toType->isTimestamp() && !fromType->isDate()) {
+  if (toType->isTimestamp() && !(fromType->isDate() || fromType->isVarchar())) {
```
Comment (JkSelf) [inline cpp/velox/substrait/SubstraitToVeloxPlanValidator.cc:265]: @PHILO-HE Why fallback the Varchar type here?

**Review by JkSelf — APPROVED** [review]
LGTM. Thanks.
